crypto/chacha/chacha_enc.c

```diff
@@ -24,6 +24,28 @@ typedef union {
 
 # define ROTATE(v, n) (((v) << (n)) | ((v) >> (32 - (n))))
 
+# ifndef PEDANTIC
+#  if defined(__GNUC__) && __GNUC__>=2 && \
+      !defined(OPENSSL_NO_ASM) && !defined(OPENSSL_NO_INLINE_ASM)
+#   if defined(__riscv_zbb) || defined(__riscv_zbkb)
+#    if __riscv_xlen == 64
+#    undef ROTATE
+#    define ROTATE(x, n) ({ int ret;                   \
```

Comment:
Should this declare `u32 ret` not `int ret`?

Note that roriw sign-extends the 32 bit word.

Reply:
> Should this declare `u32 ret` not `int ret`?

Agreed. I'll make this change to make it more clear as sign-extension should not affect the correctness (used an implicit type conversion) as you can see `x[d] = ROTATE((x[d] ^ x[a]),16)` below where `x[d]` is u32.

One more note about the sign-extension is that interestingly I found some comments on `ROTATE`, where we could make this `long` and re-arrange `QUARTERROUND` to improve performance (could be done in further PRs, should not be included here).
https://github.com/openssl/openssl/blob/4e9a4997c540e64647d4e1708a1dbda51fb59a68/include/crypto/md32_common.h#L264-L267